Add prometheus local dev container and config

alikins opened the pull request:
Add persistent volume for prometheus data.

For local dev/testing of the changes in https://github.com/ansible/galaxy-api/pull/138

Fixes: ansible/galaxy-dev#130

alikins committed:
Add prometheus local dev container and config

Add persistent volume for prometheus data

Fixes: ansible/galaxy-dev#130

cutwater reviewed (approved):
LGTM, it would be nice to have prometheus as optional component for development environment, but not immediate issue.

alikins commented:
@cutwater Whats the best way to make it optional?  

Using an optional secondary docker compose file seems to be what docker docs suggest
(https://docs.docker.com/compose/extends/#multiple-compose-files)

I'll split up the prometheus stuff to a second optional compose file.

alikins committed:
Add docker-compose-prometheus for optional prom container.

If there is a 'USE_PROM=1' env var exported or makefile cli
option provided, the Makefile will extend docker-compose.yaml
with docker-compose-prometheus.yaml and build/setup/log
an prometheus container.

alikins committed:
Use docker env file to set prometheus_multiproc_dir

Set the env var prometheus_multiproc_dir to the promentheus
data dir. This is need since prod/cloud deployments use multiple
wsgi processess.

See https://github.com/korfuri/django-prometheus/blob/master/documentation/exports.md#exporting-metrics-in-a-wsgi-application-with-multiple-processes-globally

alikins committed:
Add a grafana instance

cutwater commented:
@alikins you can simply add `prometheus` and `grafana` services to main docker-compose, but not start them by default. E.g. galaxy \ pulp services in docker-compose should not have dependencies to prometheus and grafana and `make docker/up` command should start only galaxy and pulp services.
